ballerina/http_interceptors.bal

```diff
@@ -51,3 +53,19 @@ service class DefaultErrorInterceptor {
         return alreadyBuiltErrorResponse;
     }
 }
+
+# The class used by the runtime to invoke the `decodeJwt` method to add jwt values to the request context.
+public isolated class JwtDecoder {
+
+    # Gets invoked during object initialization.
+    #
+    isolated function init() {}
+
+    # Decodes a jwt string to `[jwt:Header, jwt:Payload]`.
+    #
+    # + jwt - The jwt value as a string
+    # + return - `[jwt:Header, jwt:Payload]` if successful or else an `error`
+    isolated function decodeJwt(string jwt) returns [jwt:Header, jwt:Payload]|error {
```

Comment:
Do we define this function only to be used by the native invoke call?

Reply:
Yes. AFAIK inorder to call a ballerina method by the runtime, the method needs to reside inside a class.
https://github.com/ballerina-platform/ballerina-lang/issues/39743

Reply:
The concern is if since this is `public` class this is exposed to the user, but there is no actual use case in the user's point-of-view. Also not sure whether adding and additional async call will cause any unexpected errors. @chamil321 WDYT?

Can't we call the JWT native methods in our native code to do `jwt:decode`? This will add the jwt native jar as a dependency of http native. @ayeshLK is there a possibility to this?

Reply:
`jwt:decode` has a ballerina implementation.

Reply:
We can make the class private. Adding `public` is a mistake.

Reply:
Regarding the `async` call, since `jwt:decode` is an isolated function, it won't have any issues right?